packages/gatsby-transformer-xml/src/gatsby-node.js

```diff
@@ -1,44 +1,218 @@
-const parseXml = require(`xml-parser`)
+const convert = require(`xml-js`)
 const _ = require(`lodash`)
 
-async function onCreateNode({
-  node,
-  actions,
-  loadNodeContent,
-  createNodeId,
-  createContentDigest,
-}) {
-  const { createNode, createParentChildLink } = actions
-
-  // We only care about XML content.
+async function onCreateNode(
+  { node, actions, loadNodeContent, createNodeId, createContentDigest },
+  pluginOptions
+) {
   if (![`application/xml`, `text/xml`].includes(node.internal.mediaType)) {
     return
   }
-  const rawXml = await loadNodeContent(node)
-  const parsedXml = parseXml(rawXml)
-  const nodeArray = parsedXml.root.children.map((obj, i) => {
-    if (obj.children) {
-      obj.xmlChildren = obj.children
-      delete obj.children
+
+  // set the default options.
+  const defaultOptions = {
+    compact: true,
+    declarationKey: `declaration`,
+    instructionKey: `instruction`,
+    attributesKey: `attributes`,
+    textKey: `text`,
+    cdataKey: `cdata`,
+    doctypeKey: `doctype`,
+    parentKey: `parent`,
+  }
+
+  const {
+    useLegacyOutput,
+    nativeType,
+    trim,
+    ignoreDeclaration,
+    ignoreInstruction,
+    ignoreAttributes,
+    ignoreComment,
+    ignoreCdata,
+    ignoreDoctype,
+    ignoreText,
+  } = pluginOptions
+
+  // merge defaults with plugin options
+  const options = _.pickBy(
+    Object.assign({}, defaultOptions, {
+      nativeType: typeof nativeType !== `undefined` ? nativeType : false,
+      trim,
```

Comment:
why you for some settings are set default values and other not?

Reply:
Good question! It was mostly an assesment from my part, that the document declaration doesn’t provide too much useful data. There might be cases though. Same for vommenes. The nativeType is default set to false, to interpret output as strings, but I’m not sure if it’s better to set that to true.
The rest are set to false per default in the xml-js library, so no need for default values as undefined is falsy.